Deploy to GitHub and PyPI on tags

sethmlarson opened the pull request:
Revises our release process to:
- Create a branch named `release-[VERSION]` and update `CHANGES.rst` and `urllib3/__init__.py`.
- Open a PR against the master branch
- Integration tests run in addition to unit tests
- Once the PR is squash-merged any maintainer can tag the merge commit
- Travis builds the tagged commit and releases to PyPI and GitHub releases.
- The GitHub release is put in a draft state, requires a maintainer to approve the final draft.

Closes #1498. Closes #1392.

sethmlarson commented:
Running the Travis build again now that #1509 is merged.

theacodes commented on .travis.yml:
```diff
@@ -1,28 +1,40 @@
     - GAE_SDK_PATH=${HOME}/.cache/google_appengine
     - PYTHONWARNINGS=always::DeprecationWarning
+
+    - PYPI_USERNAME=shazow
```
Let's start a new urllib3 account, yeah? I can take care of that. This can stay as-is for this PR, though.

https://github.com/urllib3/urllib3/issues/1522

theacodes commented on _travis/deploy.sh:
```diff
@@ -0,0 +1,8 @@
+set -e
+set -x
+
+python -m pip install -U twine
```
Can you spell out the flags here? `--upgrade`.

theacodes replied:
python3?

theacodes commented on _travis/deploy.sh:
```diff
@@ -0,0 +1,8 @@
+
+python -m pip install -U twine
+python setup.py sdist bdist_wheel
+twine upload dist/* -u $PYPI_USERNAME -p $PYPI_PASSWORD --skip-existing
```
`python3 -m twine`, please.

theacodes commented on _travis/downstream/botocore.sh:
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -e
```
`set -exo pipefail`, please. (to all bash scripts)

sethmlarson commented:
I've addressed your review comments @theacodes.
From here I want to see the downstream tests run successfully once more then I'll add another commit that makes the downstream test scripts into a different optional build stage.

sethmlarson commented:
We're also waiting on requests/requests#4915 to be merged for our integration tests to pass consistently.
